### PR TITLE
Namespace modules under core/

### PR DESCRIPTION
**Pull request by mstriemer**

### CODE REVIEW

**Comment by ngokevin** [comment]
Can we also have a core/index.js or something that ```requires``` everything? So that in ```main.js``` we don't have to list every module?

File: core/init.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
+   init.js is appending to our minified bundles in the CP build process.
```
Comment (ngokevin) [inline core/init.js]: so we don't need to have the part that does ```require([settings.init_module], function(init_module) {});```?

Comment (mstriemer) [inline core/init.js]: I didn't see any reference to this being included anywhere (for fireplace at least). Shouldn't this be handled by setting the main module in requirejs anyway?

**Comment by mstriemer** [comment]
I don't see much value in an `index` module that requires everything. If there is something that needs to be initialized it is setup in the `init` module. If your app doesn't need something it shouldn't need to require it. If `main` is loading modules it isn't directly using we should stop doing that.

**Comment by ngokevin** [comment]
Well, the idea of the core modules is that everything is needed. That's why ```main.js``` has to manually load everything, and it can be a pain. It adds a step to adding a core module since you have to add it to ```main.js```'s require list to initialize it.

File: core/init.js (hunk 1 above)
Comment (ngokevin) [inline core/init.js]: It is appended by our build scripts. ```init.js``` is supposed to kick everything off when we are minified.

**Comment by ngokevin** [comment]
Yeah, nvm scratch the index.js idea

**Comment by ngokevin** [comment]
I like it. There will be changes to Commonplace (to change our RequireJS config and paths, for now) and to the build system (to specify ```main.js``` as our main module rather than ```init.js```).

**Comment by mstriemer** [comment]
I was prepping my commonplace changes and got distracted by news. I'll have that PR up by tomorrow morning at the latest. Do you know where we set `init.js` as the main module? I guess probably in `marketplace-gulp` somewhere?

**Comment by ngokevin** [comment]
Hm, it looks like I had already removed init.js from our build step and just plugged in ```main```. Nevermind then!

**Comment by mstriemer** [comment]
commonplace changes are in https://github.com/mozilla/commonplace/pull/61.

**Comment by mstriemer** [comment]
Merged into the `tracking-2.0.0` branch with 75e7919.

**Pull request closed without merging**